tests/src/test/java/alluxio/collections/IndexedSetConcurrencyTest.java

```diff
@@ -76,6 +78,25 @@ public long runSingleTask() {
     }
   }
 
+  private class ConcurrentAddWithCheck extends ConcurrentTask {
+    @Override
+    public long runSingleTask() {
+      long result = 0;
```

Comment:
Do you mean we use the barrier to let all the threads wait and then start together?